packages/client-cli/src/main.rs

```diff
@@ -76,6 +76,16 @@ enum Subcommands {
     }
 }
 
+fn check_is_bin(mut file_path: PathBuf) {
+    //file_path = file_path.canonicalize().unwrap();
+    assert_eq!("bin", file_path.extension().unwrap_or(OsStr::new("bin")), "{} should have .bin extension", file_path.display());
+    let mut buffer = [0 as u8; 4];
+    let n = File::open(&file_path).expect(&*format!("{} does not exist", file_path.display())).read(&mut buffer[..]);
+    let buffer = buffer;
+    let elf_header = [0x7F as u8, 'E' as u8, 'L' as u8, 'F' as u8];
+    assert_ne!(buffer, elf_header, "{} is an elf file, use a bin file instead", file_path.display());
+}
+
```

Comment:
We should not check file extensions because there are cases when you may want to run the simulator on a file that doesn't have a `.bin` extension. Instead, can we, for example, check for the XVX5 magic? Also, I think this should return an error rather than panicking since panicking usually implies an issue with the program rather than an issue with the program's input.